consensus/propagation/reactor.go

```diff
@@ -299,14 +313,160 @@ func chunkIndexes(totalSize, chunkSize int) [][2]int {
 // - otherwise, we send the wants for the missing parts to that peer before broadcasting the haves.
 // - finally, we keep track of the want requests in the proposal state.
 func (blockProp *Reactor) handleHaves(peer p2p.ID, haves *proptypes.HaveParts, bypassRequestLimit bool) {
+	if haves == nil {
+		return
+	}
```

Comment:
might eventually want to through and error and potentially disconnect etc if this occurs

fine for now ofc